Better handle audio and video conversion

MonoS opened the pull request:
Those are the changes made and the reasons:
avi:rgb => avi:jyuv: as the documentations says "This is most similar to the original PlayStation video colors"
add zscale: the PSX video are in a particular color space (full range, bt601, centered chroma location), but modern browser, AFAIK, will use the "full HD colorspace" (limited range, bt709, left chroma location), making this conversion at the encoding stage will make sure that browser will display correct colors and levels
w=582:h=448: upscale the video and fix AR, i'm not quite sure regarding the AR, i've tested it counting pixel from duckstation on a couple of videos
-crf 15: a bit of a quality boost, the videos are quite small anyway, it can be raised to lower quality/size
-preset veryslow: quality boost at the expense of time, the video are short and at low resolution/framerate, it will take not much more
-level 31: force H264 level to 3.1 to allow better compatibility
-ar 44100: with default conversion ffmpeg was downsampling the 18.9KHz sound to 16KHz, also AAC use an internal lowpass filter so it was muffling the sound, this will upsample to 44.1KHz so that it will sound correctly
-b:a 192k: a bit more bitrate for the sound, it can probably be lowered to 128k but i prefer a bit more as the original sounds are already quite compressed

Regarding the videos a better job could be made introducing avs/vs in the mix, the original videos suffers quite a bit of blocking and Deblock_QED fix almost all of it with minimal/low loss of quality, also the upscale can be done using better algorithms like nnedi3 or a specifically trained ESRGAN model

MonoS commented:
Here an example of video INS17.STR[0]

This is how it is converted right now
https://user-images.githubusercontent.com/801011/227200980-572c391a-14a1-4dfc-ba12-e6cde68338fe.mp4

This is with my settings
https://user-images.githubusercontent.com/801011/227201022-4127bc12-2878-4e25-aed4-2e090c6ad4ab.mp4

This instead is an experiment i made using ESRGAN and model 4xFSDedither_Manga taken from here https://upscale.wiki/wiki/Model_Database#Dithering (i haven't tried other in reality) 
https://user-images.githubusercontent.com/801011/227201211-1a572989-6e56-4a4b-b4ee-b0c704ff0fc1.mp4

elliotcraft79 commented:
What benefit is there to upscaling the video using ffmpeg over keeping it at the original resolution? I have checked the game and the aspect ratio and resolution of the original video is correct. avi:rgb is used as it provides the most accurate colours. The rest of the PR is good, especially the audio stuff (it's kinda strange that ffmpeg did that).

elliotcraft79 commented:
The game itself initialises the screen at 320x240, so the aspect ratio is definitely correct (duckstation uses 319x224 instead of 320x240 for some reason).
![image](https://user-images.githubusercontent.com/16109561/227716051-ca98d6e8-a8d0-4880-945f-b3ecb56b0a8d.png)
Chromium (and probably firefox) does indeed appear to use bilinear scaling, but I don't think it would matter if we left them as is (especially with the already poor quality the videos are in).
As for avi:rgb vs avi:jyuv, it would probably be better to ask the dev of jpsxdec as you said.

MonoS commented:
>The game itself initialises the screen at 320x240, so the aspect ratio is definitely correct

I'll ask the devs of jpsxdec for that also, as the preview windows is not 320*240, but i'll amend my PR to not change the resolution

>but I don't think it would matter if we left them as is (especially with the already poor quality the videos are in).

I think the same, my upscale was used only to fix the AR

>As for avi:rgb vs avi:jyuv

Let me know, personally i'll leave jyuv, but if you intend to accept this PR in the near future and prefer rgb i can revisit the colorpsace conversion to start with an rgb input

MonoS committed:
Raise audio sample rate in extract_media.mjs

The PSX audio is at 18.9KHz but ffmpeg was downsampling it to 16KHz muffling the sound, force the audio to be sampled at 44.1KHz so that it sounds correctly.
